Windows で file mode (100644/100755) が検出されないテスト失敗

## 概要

Windows では `fs.stat()` が POSIX の executable bit を返さないため、`100644` ↔ `100755` の mode 変更を正しく検出できず、関連テスト ~20件が失敗する。

サンプル:
```
AssertionError: '' == 'diff --git a/file.txt b/file.txt\nold mode 100644\nnew mode 100755'
```

## 失敗するテスト(代表例)

- `diff.test.ts > index/workspace > ファイルモードに変換があるとき、モードのdiffを表示する`
- `diff.test.ts > head/index > ファイルモード...同様`
- `merge.test.ts > conflicted merge: add-add mode conflict > ...`
- `status.test.ts` の一部
- ほか mode 比較が絡むケース全般

## 背景

git 本家の Windows 対応:
- `core.filemode` 設定(`true`/`false`)で「mode を本当に追跡するか」を切り替え
- Windows のデフォルトは `core.filemode=false` で、`fs.stat` 由来の mode 情報を信頼しない
- index 上は最初に追加された時の mode をそのまま保持し、以後 mode 変更は無視

参考: https://git-scm.com/docs/git-config#Documentation/git-config.txt-corefileMode

## 修正の方向性

1. `Repository.config.get(["core", "fileMode"])` を読む経路を追加
2. `Inspector.compareIndexToWorkspace` 等の mode 比較を、`core.filemode=false` のときスキップ
3. デフォルト値を OS で出し分け(Windows なら false, それ以外は true)
4. テスト側でも `core.filemode=false` 前提のケースは Windows で期待値を変えるか skip

## スコープ

- 既存 unit/integ テストへの影響を最小化しつつ、Windows での mode 系テストを通す
- Linux/macOS の挙動はそのまま維持

--- a/integ/command/add.test.ts
+++ b/integ/command/add.test.ts
@@ -1,9 +1,8 @@
 import { stripIndent } from "../../src/util";
 import * as T from "./helper";
+import { itOnlyUnix } from "./helper";
 
 const t = T.create();
-
-const itOnlyUnix = process.platform === "win32" ? it.skip : it;
 
 describe("add", () => {
   async function assertIndex(expected: [number, string][]) {

--- a/integ/command/checkout.test.ts
+++ b/integ/command/checkout.test.ts
@@ -1,6 +1,7 @@
 import assert from "node:assert";
 import { stripIndent } from "../../src/util";
 import * as T from "./helper";
+import { itOnlyUnix } from "./helper";
 const t = T.create();
 
 describe("checkout", () => {
@@ -87,7 +88,7 @@ describe("checkout", () => {
       assertStaleFile("1.txt");
     });
 
-    it("fails to update a modified-mode file", async () => {
+    itOnlyUnix("fails to update a modified-mode file", async () => {
       await t.writeFile("1.txt", "changed");
       await commitAll();
 
@@ -146,7 +147,7 @@ describe("checkout", () => {
       await assertStatus("");
     });
 
-    it("failes to update a staged changed-mode file", async () => {
+    itOnlyUnix("failes to update a staged changed-mode file", async () => {
       await t.writeFile("1.txt", "changed");
       await commitAll();
 
@@ -293,7 +294,7 @@ describe("checkout", () => {
       assertStaleFile("outer/94.txt");
     });
 
-    it("fails to remove a changed-mode file", async () => {
+    itOnlyUnix("fails to remove a changed-mode file", async () => {
       await t.writeFile("outer/94.txt", "94");
       await commitAll();
 
@@ -339,7 +340,7 @@ describe("checkout", () => {
       assertStaleFile("outer/94.txt");
     });
 
-    it("fails to remove a staged changed-mode file", async () => {
+    itOnlyUnix("fails to remove a staged changed-mode file", async () => {
       await t.writeFile("outer/94.txt", "94");
       await commitAll();
 

--- a/integ/command/diff.test.ts
+++ b/integ/command/diff.test.ts
@@ -1,4 +1,5 @@
 import * as T from "./helper";
+import { itOnlyUnix } from "./helper";
 import { stripIndent } from "../../src/util";
 import { Repository } from "../../src/repository";
 
@@ -38,21 +39,26 @@ describe("diff", () => {
       `);
     });
 
-    it("ファイルモードに変換があるとき、モードのdiffを表示する", async () => {
-      await t.makeExecutable("file.txt");
+    itOnlyUnix(
+      "ファイルモードに変換があるとき、モードのdiffを表示する",
+      async () => {
+        await t.makeExecutable("file.txt");
 
-      await assertDiff(stripIndent`
+        await assertDiff(stripIndent`
         diff --git a/file.txt b/file.txt
         old mode 100644
         new mode 100755
       `);
-    });
+      },
+    );
 
-    it("コンテンツ・ファイルモードに変換があるとき、コンテンツ・モードのdiffを表示する", async () => {
-      await t.writeFile("file.txt", "changed");
-      await t.makeExecutable("file.txt");
+    itOnlyUnix(
+      "コンテンツ・ファイルモードに変換があるとき、コンテンツ・モードのdiffを表示する",
+      async () => {
+        await t.writeFile("file.txt", "changed");
+        await t.makeExecutable("file.txt");
 
-      await assertDiff(stripIndent`
+        await assertDiff(stripIndent`
         diff --git a/file.txt b/file.txt
         old mode 100644
         new mode 100755
@@ -63,7 +69,8 @@ describe("diff", () => {
         -contents
         +changed
       `);
-    });
+      },
+    );
 
     it("ファイルが削除されたとき、削除のdiffを表示する", async () => {
       await t.rm("file.txt");

--- a/integ/command/helper.ts
+++ b/integ/command/helper.ts
@@ -19,6 +19,18 @@ export function create(name?: string) {
   return new TestUtil(name);
 }
 
+/**
+ * Windows では fs.chmod / fs.stat が POSIX の executable bit を扱えないため、
+ * ファイルモード(100644 / 100755)や読み取り権限など POSIX 固有の挙動に
+ * 依存するテストは実行しても確実に失敗する。
+ * 本来は core.filemode = false 相当の動作を実装すべきだが、現状は skip する。
+ *
+ * 関連: issue #20 (Windows file mode 未検出)
+ */
+const isWindows = process.platform === "win32";
+export const itOnlyUnix = isWindows ? it.skip : it;
+export const describeOnlyUnix = isWindows ? describe.skip : describe;
+
 const fs = promises;
 export type Contents = [string, string][];
 interface MockTTYOptions {

--- a/integ/command/merge.test.ts
+++ b/integ/command/merge.test.ts
@@ -1,4 +1,5 @@
 import * as T from "./helper";
+import { describeOnlyUnix } from "./helper";
 import assert from "node:assert";
 import { Pathname } from "../../src/types";
 import { stripIndent } from "../../src/util";
@@ -390,7 +391,7 @@ describe("merge", () => {
     });
   });
 
-  describe("conflicted merge: add-add mode conflict", () => {
+  describeOnlyUnix("conflicted merge: add-add mode conflict", () => {
     beforeEach(async () => {
       // prettier-ignore
       await t.merge3(

--- a/integ/command/status.test.ts
+++ b/integ/command/status.test.ts
@@ -1,4 +1,5 @@
 import * as T from "./helper";
+import { itOnlyUnix } from "./helper";
 import { stripIndent } from "../../src/util";
 
 const t = T.create();
@@ -133,7 +134,7 @@ describe("status", () => {
       `);
     });
 
-    it("reports files with change modes", async () => {
+    itOnlyUnix("reports files with change modes", async () => {
       // Arrange
       await t.makeExecutable("a/2.txt");
 
@@ -208,7 +209,7 @@ describe("status", () => {
       await assertStatusPorcelain("A  d/e/5.txt");
     });
 
-    it("reports modified modes", async () => {
+    itOnlyUnix("reports modified modes", async () => {
       await t.makeExecutable("1.txt");
       await t.kitCmd("add", ".");
 

